core/src/main/java/org/infinispan/functional/impl/AbstractFunctionalMap.java

```diff
@@ -150,4 +160,18 @@ private Transaction getOngoingTransaction() {
          return cf;
       }
    }
+
+   protected Set encodeKeys(Set<? extends K> keys) {
+      return keys.stream().map(k -> keyDataConversion.toStorage(k)).collect(Collectors.toSet());
```

Comment:
I wonder if we want to convert this to use something similar to https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/cache/impl/EncoderCache.java#L87